proposals/ref-assignment-switch-expressions.md

```diff
@@ -0,0 +1,110 @@
+# `ref` assignment on `switch` expressions
+
+* [x] Proposed
+* [ ] Prototype: None
+* [ ] Implementation: None
+* [ ] Specification: Started, below
+
+## Summary
+[summary]: #summary
+
+Provide the ability to return `ref` values in `switch` expressions.
+
+## Motivation
+[motivation]: #motivation
+
+The `switch` expression is meant to provide a quicker way to return values based on a given expression, but does not support returning `ref` expressions. This limitation doesn't prevent any harm, and can be lifted.
+
+## Detailed design
+[design]: #detailed-design
+
+A `switch` expression may return a `ref` value. The returning expression, if not a `throw` expression, must be a `ref` expression.
+
+A `switch` expression that returns a `ref` value needs an extra `ref` in front of it when assigned/returned to a `ref` local. This design aligns with the current design in the ternary operator.
+
+`throw` expressions can still be used normally, and pose no effect on the return type of the `switch` expression.
+
+Once a single switch arm returns a `ref` expression, all other switch arms must return `ref` expressions, unless they are `throw` expressions. Not providing a `ref` expression in a switch arm causes a compiler error.
+
+Since all ref expressions are LValues, and the switch arms are all ref expressions, the entire result of the switch expression is also an LValue. Therefore, the entire switch expression can be passed by reference, assigned to, and returned by reference. Note that for directly assigning to the switch expression's ref result, the `ref` keyword must not be used, like normal ref locals are assigned. For example:
+
+`item switch { 1 => ref X, 2 => ref Y } = value`
```

Comment:
Consider using a code block for better syntax highlighting.

````suggestion
```cs
item switch { 1 => ref X, 2 => ref Y } = value
```
````